include/deal.II/base/utilities.h

```diff
@@ -1307,7 +1307,8 @@ namespace Utilities
 #else
         (void)allow_compression;
 #endif
-        fisb.push(boost::iostreams::array_source(&*cbegin, &*cend));
+        fisb.push(
+          boost::iostreams::array_source(&*cbegin, &*cbegin + (cend - cbegin)));
```

Comment:
Example 34.3 of https://theboostcpplibraries.com/boost.iostreams-devices suggests that we could write this as
```suggestion
          boost::iostreams::array_source(&*cbegin, cend - cbegin));
```
Indeed, this overload exists: https://www.boost.org/doc/libs/1_76_0/libs/iostreams/doc/classes/array.html

Reply:
Thanks @bangerth , I applied it.